scripts/unity/header_prepare.py

```diff
@@ -13,19 +13,26 @@ def header_prepare(in_file, out_file, out_wrap_file):
         content = f_in.read()
 
     # remove comments
-    comments_pattern = re.compile(
-        r'//.*?$|/\*.*?\*/|\'(?:\\.|[^\\\'])*\'|"(?:\\.|[^\\"])*"',
+    c_comments_pattern = re.compile(
+        r'/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*+/',
         re.DOTALL | re.MULTILINE
     )
-    content = comments_pattern.sub(r"", content)
+
+    content = c_comments_pattern.sub(r"", content)
+
+    cpp_comments_pattern = re.compile(
+        r'(?!\".?)\/\/\w+$',
```

Comment:
The `$` in this case is not working as expected:
```
with open('test.txt') as f_in:
    content = f_in.read()
content = c_comments_pattern.sub(r"", content)
content = cpp_comments_pattern.sub(r"", content)

print(content)
...
#include "foo_internal.h"
// c++ comment
...
```

This alse means that comments following code, such as:
```
int foo_init(void *handle); // foo_init declaration
```
are also not removed correctly.

Reply:
right, can you take another look? Changed to:

```
cpp_comments_pattern = re.compile(
        r'(?!.*\".*)\/\/.*$',
        re.DOTALL | re.MULTILINE
    )
```

Reply:
Still not working when loading a file into python shell, and testing it there.
Wonder if the `re.MULTILINE` could be impacting the `$` in the regex in this case.

Reply:
removed `re.DOTALL` and that helped. Please check again.

Reply:
It sure did :+1: